src/components/ResetPassword/index.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import ImageFader from '../Common/ImageFader';
+import { defineMessages, useIntl } from 'react-intl';
+import LanguagePicker from '../Layout/LanguagePicker';
+import Button from '../Common/Button';
+import { Field, Form, Formik } from 'formik';
+import * as Yup from 'yup';
+import axios from 'axios';
+import { useRouter } from 'next/router';
+
+const messages = defineMessages({
+  resetpassword: 'Reset your password',
+  password: 'Password',
+  confirmpassword: 'Confirm Password',
+  validationpasswordrequired: 'Password required',
+  validationconfirmpasswordrequired: 'Confirm Password required',
+  validationpasswordmatch: 'Password must match',
+  gobacklogin: 'Go Back to Login Page',
+  successmessage:
+    'If the link is valid and is connected to a user then the password has been reset.',
+});
+
+const ResetPassword: React.FC = () => {
+  const intl = useIntl();
+  const router = useRouter();
+  const [hasSubmitted, setSubmitted] = useState(false);
+
+  const guid = router.query.guid;
+
+  const ResetSchema = Yup.object().shape({
+    password: Yup.string()
+      .required(intl.formatMessage(messages.validationpasswordrequired))
+      .min(8),
```

Comment:
Add a validation message for the minimum password length? "Password is too short; should be a minimum of 8 characters" (same as in the create local user modal)